src/main/java/org/opensearch/ad/transport/EntityResultTransportAction.java

```diff
@@ -158,14 +162,14 @@ private ActionListener<Optional<AnomalyDetector>> onGetDetector(
     ) {
         return ActionListener.wrap(detectorOptional -> {
             if (!detectorOptional.isPresent()) {
-                listener.onFailure(new EndRunException(detectorId, "AnomalyDetector is not available.", true));
+                listener.onFailure(new EndRunException(detectorId, "AnomalyDetector is not available.", false));
                 return;
             }
 
             AnomalyDetector detector = detectorOptional.get();
 
             if (request.getEntities() == null) {
-                listener.onResponse(null);
+                listener.onFailure(new EndRunException(detectorId, "Fail to get any entities from request.", false));
```

Comment:
For sparse data, it's possible that some interval has no data. With this change, we will stop AD job if there are no data for X intervals?

Reply:
if there is no data, model node should not receive the request in the first place. If request.getEntities() == null, there might be some bug in our code.